modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesMessageMetadata.java

```diff
@@ -108,8 +108,24 @@
      */
     private Map<String, Object> propertyMap;
 
+    /**
+     * True if this message should be retained
+     */
+    private boolean retain;
```

Comment:
please add a more description and explain following:
1) Whats retain ( definition of retain).
2) Whats the applicable use cases ( e.g. this concept origins from MQTT)